### PR TITLE
Age Filter Teacher Override

### PR DESCRIPTION
**Pull request by epeach**
Issues: https://github.com/code-dot-org/dance-party/issues/351

Check url for override flag before any other render. If flag is present, set session flag similar to setting age in dialog. (Now even if redirected to page without flag in url, flag stays on for entire session)

Removes url checks from other places (will hit new check in AgeDialog before others)

Includes Tests

### CODE REVIEW

File: apps/src/dance/DanceVisualizationColumn.jsx [hunk 1]
```diff
@@ -72,10 +71,10 @@ class DanceVisualizationColumn extends React.Component {
   };
 
   /*
-    Turn the song filter off unless there is a teacher override
+    Turn the song filter off
   */
   turnFilterOff() {
-    this.setState({filterOff: queryString.parse(window.location.search).songfilter !== 'on'});
+    this.setState({filterOff: true});
```
Comment (epeach) [inline apps/src/dance/DanceVisualizationColumn.jsx:77]: queryString check is no longer necessary - we pass this function to AgeDialog and the first thing we do in AgeDialog is check this.

File: apps/src/templates/AgeDialog.jsx [hunk 2]
```diff
@@ -88,8 +81,31 @@ class AgeDialog extends Component {
...
+    }
+  };
+
   render() {
     const { signedIn} = this.props;
+
+    // If the song filter override has been turned on, set session storage
+    // Dialog won't render
+    if (queryString.parse(window.location.search).songfilter === 'on') {
+      this.setManualFilter();
```
Comment (epeach) [inline apps/src/templates/AgeDialog.jsx]: Will set sessionStorage.getItem(sessionStorageKey), so will return null in following condition.

File: dashboard/test/ui/features/dance/age_filter.feature [hunk 3]
```diff
@@ -81,3 +81,47 @@ Feature: Dance Lab Age Filter
...
+    #Run for two measures until level success
+    Then I click selector "#runButton" once I see it
+    And I wait until element ".congrats" is visible
+    And I press "continue-button"
+    # Make sure continue takes us to next level
+    And I wait until current URL contains "/stage/37/puzzle/2"
+    And I wait for the page to fully load
+    And I close the instructions overlay if it exists
+    #Local PG-13 option should not be visible after filter in any environment
+    And I do not see "synthesize" option in the dropdown "#song_selector"
```
Comment (davidsbailey) [inline dashboard/test/ui/features/dance/age_filter.feature:107]: FYI, you should be able to see the restricted song list locally (and in circleci) now if you have s3 and cloudfront keys set.

File: apps/src/templates/AgeDialog.jsx [hunk 4]
```diff
@@ -88,8 +81,31 @@ class AgeDialog extends Component {
...
 
+  onClickAgeOk = () => {
+    const value = this.ageDropdown.getValue();
+    // Ignore click if nothing selected
+    if (!value) {
+      return;
+    }
+
+    // Sets cookie to true when anon user is 13+. False otherwise.
+    let over13 = parseInt(value, 10) >= 13;
```
Comment (Bjvanminnen) [inline apps/src/templates/AgeDialog.jsx]: supernit: could be const?

**Pull request merged**